pkg/forwarder/forwarder.go

```diff
@@ -385,7 +386,14 @@ func (f *DefaultForwarder) setAPIKeyStatus(apiKey string, domain string, status
 func (f *DefaultForwarder) validateAPIKey(apiKey, domain string) (bool, error) {
 	url := fmt.Sprintf("%s%s?api_key=%s", config.Datadog.GetString("dd_url"), v1ValidateEndpoint, apiKey)
 
-	resp, err := http.Get(url)
+	transport := util.CreateHTTPTransport()
+
+	client := &http.Client{
+		Transport: transport,
+		Timeout:   5 * time.Second,
```

Comment:
any reason to choose a timeout of 5 seconds here? If so let's add an inline comment to explain why.

Otherwise we may want to pick the same timeout value as the forwarder's `forwarder_timeout` (default of 20 seconds)

Reply:
addressed in https://github.com/DataDog/datadog-agent/pull/1601